Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1000,66 +1062,13 @@ BoardInit (
     }
     break;
   case ReadyToBoot:
-
-    SiliconCfgData = (SILICON_CFG_DATA *)FindConfigDataByTag (CDATA_SILICON_TAG);
-    if ( (SiliconCfgData != NULL) && (SiliconCfgData->ECEnable == 1)){
-      //
-      // Enable decoding of I/O locations 62h and 66h to LPC
-      //
-      LpcBase = MmPciBase (0, PCI_DEVICE_NUMBER_PCH_LPC, 0);
-      MmioOr16 (LpcBase + R_LPC_CFG_IOE, B_LPC_CFG_IOE_ME1);
-
-      //
-      // Enable EC's ACPI mode to control power to motherboard during Sleep (S3)
-      //
-      IoWrite16 (EC_C_PORT, EC_C_ACPI_ENABLE);
-    }
-
-    // Lock down SPI for everything
-    SpiBaseAddress = GetDeviceAddr (OsBootDeviceSpi, 0);
-    SpiBaseAddress = TO_MM_PCI_ADDRESS (SpiBaseAddress);
-
-    // Set the BIOS Lock Enable and EISS bits
-    MmioOr8 (SpiBaseAddress + R_SPI_CFG_BC, (UINT8) (B_SPI_CFG_BC_LE | B_SPI_CFG_BC_EISS));
-
-    //
-    // Set Bios Interface Lock-Down (BILD)
-    // B0:D31:F05  Offset DCh Bit 07
-    //
-
-    // Set the BILD
-    MmioOr8 (SpiBaseAddress + R_SPI_CFG_BC, (UINT8)B_SPI_CFG_BC_BILD);
-
-    //
-    // set Flash Configuration Lock-Down (FLOCKDN)
-    // B0:D31:F05  Offset 04h Bit 15
-    //
-    SpiBar0 = (UINTN)PciRead32 (PCI_LIB_ADDRESS (0, PCI_DEVICE_NUMBER_PCH_SPI, PCI_FUNCTION_NUMBER_PCH_SPI, R_SPI_CFG_BAR0));
-    SpiBar0 &= ~(B_SPI_CFG_BAR0_MASK);
-    if (SpiBar0 != 0) {
-      MmioOr16 (SpiBar0 + R_SPI_MEM_HSFSC, (UINT16)B_SPI_MEM_HSFSC_FLOCKDN);
-    }
-
-    //
-    // set SMI LOCK (SMI_LOCK)
-    // B0:D31:F02  Offset 1024h Bit 4
-    //
-    MmioOr8 (PCH_PWRM_BASE_ADDRESS + R_PMC_PWRM_GEN_PMCON_B, (UINT8)B_PMC_PWRM_GEN_PMCON_B_SMI_LOCK);
-
-    ClearFspHob ();
-
-    //
-    // Lock down Tco WDT just before handling off to OS
-    //
-    TcoBase = MmioRead16(PCH_PCR_ADDRESS(PID_DMI, R_PCH_DMI_PCR_TCOBASE)) & B_PCH_DMI_PCR_TCOBASE_TCOBA;
-    IoOr16 ((TcoBase + R_TCO_IO_TCO1_CNT), B_TCO_IO_TCO1_CNT_LOCK);
-
-    break;
-  case EndOfFirmware:
     ClearSmi ();
     if ((GetBootMode() == BOOT_ON_S3_RESUME) && (GetPayloadId () == UEFI_PAYLOAD_ID_SIGNATURE)) {
       RestoreS3RegInfo (FindS3Info (S3_SAVE_REG_COMM_ID));
     }
+    ProcessAllLocks ();
```

Comment:
If flash is locked, variable could not work in Osloader.
Maybe rename it to ProgramSecuritySetting(). and  do it here or EndOfFirmware based on PlatformId.